### ISSUE TITLE
Optional field image and 'childImageSharp' as null | Gatsby, GraphQL & YAML

### ISSUE THREAD

**Issue by ItzaMi** [gatsbyjs/gatsby#22857]
I'm sorry in advance if this issue was treated somewhere else but I couldn't find an answer to what I need. 

I'm currently working with Gatsby, GraphQl and YAML files and I have multiple sections that I'm mapping over but some have images and some don't.  
I defined my GraphQL by stating the image with childImageSharp

```
sections {
              title
              description
              imageHere {
                childImageSharp {
                  fluid(maxWidth: 600) {
                    ...GatsbyImageSharpFluid_noBase64
                  }
                }
              }
```

and I saw a solution that involved adding this to my gatsby-node.js so that the paths could be properly read from my YAML files

```
exports.createSchemaCustomization = ({ actions }) => {
  const { createTypes } = actions;
  const typeDefs = `
    type allProjectPageYaml implements Node {
      imageHere: [String]!
    }
  `;
  createTypes(typeDefs);
};
```

Now my problem is that some sections have the imageHere field and some don't
```
sections:
 - title: Title 1
   description: Description 1
 - title: Title 2
   description: Description 2
   imageHere: Image1
```

The error I get is **TypeError: Cannot read property 'childImageSharp' of null**

What's my workaround this?

**Comment by ascorbic** [comment]
Hi. Is the error in your component? If so, the answer would be to add a conditional. e.g. 

```jsx

export default const MyTemplateComponent = ({ data }) => {
    const page = data.projectPageYaml;
    return (
      <div><h1>{page.title}</h1>{page.imageHere && <Image fixed={page.imageHere.childImageSharp.fixed} />}</div>
   )
}
//Page query here
```

I'm goign to close this, but feel free to reopen if this doesn't answer your question.

**Comment by ItzaMi** [comment]
Hi, @ascorbic , thanks for your reply but it seems like that doesn't properly help me.  
  
After transfering this page to a template (since the proposition is to have multiple pages created the same way), I ran into another problem.  
childImageSharp is taking my image path and just turning it into a string (I opened another issue [here](https://github.com/ItzaMi/childImageSharp-issue) regarding this but I'm not sure if my reproduction repo properly explains what's happening) and now I get 'TypeError: Cannot read property 'fluid' of undefined' from any image in this pages.

Any idea what is making this?
